services/galley/src/Galley/API/Teams.hs

```diff
@@ -905,9 +908,39 @@ setFeatureStatus setter doauth tid status = do
       zusrMembership <- Data.teamMember tid uid
       void $ permissionCheck (ChangeTeamFeature (Public.knownTeamFeatureName @a)) zusrMembership
     DontDoAuth ->
-      pure ()
+      assertTeamExists tid
   setter tid status
 
+getAllFeaturesH :: UserId ::: TeamId ::: JSON -> Galley Response
+getAllFeaturesH (uid ::: tid ::: _) =
+  json <$> getAllFeatures uid tid
+
+getAllFeatures :: UserId -> TeamId -> Galley Aeson.Value
+getAllFeatures uid tid = do
+  Aeson.object
+    <$> sequence
+      [ w @'Public.TeamFeatureSSO getSSOStatusInternal,
+        w @'Public.TeamFeatureLegalHold getLegalholdStatusInternal,
+        w @'Public.TeamFeatureSearchVisibility getTeamSearchVisibilityAvailableInternal,
+        w @'Public.TeamFeatureValidateSAMLEmails getValidateSAMLEmailsInternal,
+        w @'Public.TeamFeatureDigitalSignatures getDigitalSignaturesInternal,
+        w @'Public.TeamFeatureAppLock getAppLockInternal
+      ]
+  where
+    w ::
+      forall (a :: Public.TeamFeatureName).
+      ( Public.KnownTeamFeatureName a,
+        Aeson.ToJSON (Public.TeamFeatureStatus a)
+      ) =>
+      ( TeamId ->
+        Galley (Public.TeamFeatureStatus a)
+      ) ->
+      Galley (Text, Aeson.Value)
+    w getter = do
+      status <- getFeatureStatus @a getter (DoAuth uid) tid
+      let feature = Public.knownTeamFeatureName @a
+      pure $ (cs . toByteString' $ feature, Aeson.toJSON status)
```

Comment:
```suggestion
      pure $ (feature Aeson..= status)
```

(This requires a `KeyValue` instance for `TeamFeatureName`.)

Reply:
applied your suggestion partly. Looks like first argument to `.=` is not overloaded, but must be `Text`